pkg/engine/vulnerability_builder_test.go

```diff
@@ -298,12 +337,52 @@ func Test_detectLine(t *testing.T) {
 				},
 				searchKey: "aws_s3_bucket[b].Environment={{Dev.123}}",
 			},
-			want: 7,
+			want: vulnerabilityLines{
+				line: 7,
+				vulnLine: model.VulnLines{
+					Positions: []int{6, 7, 8},
+					Lines: []string{
+						`						  Name        = "My bucket"`,
+						`						  Environment = "Dev.123"`,
+						`						  Environment = "test"`,
+					},
+				},
+			},
+		},
+		{
+			name: "detect_line_error",
+			args: args{
+				ctx: &QueryContext{
+					scanID: "scanID",
+				},
+				file: &model.FileMetadata{
+					ScanID: "scanID",
+					ID:     "Test",
+					Kind:   model.KindTerraform,
+					OriginalData: `resource "aws_s3_bucket" "b" {
+						bucket = "my-tf-test-bucket"
+						acl    = "authenticated-read"
+
+						tags = {
+						  Name        = "My bucket"
+						  Environment = "Dev.123"
+						  Environment = "test"
+						}
+					      }
+					      `,
+				},
+				searchKey: "testing.error",
+			},
+			want: vulnerabilityLines{
+				line:     -1,
+				vulnLine: model.VulnLines{},
+			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := detectLine(tt.args.file, tt.args.searchKey, &zerolog.Logger{}); got != tt.want {
+			got := detectLine(tt.args.file, tt.args.searchKey, &zerolog.Logger{}, 3)
+			if !reflect.DeepEqual(got, tt.want) {
 				t.Errorf("detectLine() = %v, want %v", got, tt.want)
```

Comment:
use `test.StringifyStruct` to make the output easier to inspect in case of failure

Reply:
resolved